packages/router/src/statemanager/state_manager.ts

```diff
@@ -66,21 +79,82 @@ export abstract class StateManager {
    * location change listener due to a URL update by the AngularJS router. In this case, the router
    * still need to know what the browser's URL is for future navigations.
    */
-  abstract getRawUrlTree(): UrlTree;
+  getRawUrlTree(): UrlTree {
+    return this.rawUrlTree;
+  }
 
-  /** Returns the current state stored by the browser for the current history entry. */
-  abstract restoredState(): RestoredState | null | undefined;
+  protected createBrowserPath(currentTransition: Navigation): string {
+    const rawUrl =
+      currentTransition.finalUrl !== undefined
+        ? this.urlHandlingStrategy.merge(currentTransition.finalUrl!, currentTransition.initialUrl)
+        : currentTransition.initialUrl;
+    const url = currentTransition.targetBrowserUrl ?? rawUrl;
```

Comment:
nit: since the build tooling does not minify properties, we can update the code to reference this like this instead for code optimization purposes: 
```suggestion
    const {finalUrl, initialUrl} = currentTransition;
    const rawUrl =
      finalUrl !== undefined
        ? this.urlHandlingStrategy.merge(finalUrl, initialUrl)
        : initialUrl;
    const url = currentTransition.targetBrowserUrl ?? rawUrl;
```
There are some similar cases below as well.